pkg/api/rest/rest.go

```diff
@@ -62,7 +62,7 @@ type Lister interface {
 	// This object must be a pointer type for use with Codec.DecodeInto([]byte, runtime.Object)
 	NewList() runtime.Object
 	// List selects resources in the storage which match to the selector.
-	List(ctx api.Context, label labels.Selector, field fields.Selector) (runtime.Object, error)
+	List(ctx api.Context, label labels.Selector, field fields.Selector, options *api.ListOptions) (runtime.Object, error)
```

Comment:
Separate PR is fine for me, it's mostly busywork refactor.

We have the constructor for delete options because it's impossible to do
"&0" (take address of constant) in Go.  Case is weaker for other value
types.

On Mon, Oct 26, 2015 at 3:52 PM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/api/rest/rest.go
> <https://github.com/kubernetes/kubernetes/pull/16273#discussion_r43043374>
> :
>
> > @@ -62,7 +62,7 @@ type Lister interface {
> >  	// This object must be a pointer type for use with Codec.DecodeInto([]byte, runtime.Object)
> >  	NewList() runtime.Object
> >  	// List selects resources in the storage which match to the selector.
> > -	List(ctx api.Context, label labels.Selector, field fields.Selector) (runtime.Object, error)
> > +	List(ctx api.Context, label labels.Selector, field fields.Selector, options *api.ListOptions) (runtime.Object, error)
>
> Yes - definitely I was to move label and field into listOptions (here and
> in client, etc.), but I would like to do it in a separate PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16273/files#r43043374>.
>